src/ol/source/tileimagesource.js

```diff
@@ -191,3 +198,139 @@ ol.source.TileImage.prototype.useTile = function(z, x, y) {
+  // it means we end up skipping some load requests entirely if they were
+  // immediately succeeded by another request for the same tile
+  // if this function is called by a UI slider (which can generate a huge
+  // number of requests in a short time), this delay can offer a significant
+  // speedup
+  var afterDelay = function() {
+    this.executeLoadRequests(map, z, tileRange, viewState.projection);
+  };
+  window.setTimeout(goog.bind(afterDelay, this), 100);
+};
+
+
+/**
+ * run the queued out-of-band load requests
+ *
+ * @param {ol.Map} map A map instance.
+ * @param {number} z The zoom.
+ * @param {ol.TileRange} tileRange A range of tiles.
+ * @param {ol.proj.Projection} projection The map view projection.
+ */
+ol.source.TileImage.prototype.executeLoadRequests =
+    function(map, z, tileRange, projection) {
+
+  var x, y, tileCoordKey;
+
+  if (this.tileCache.getCount() > 0) {
+    // clear any tiles that are not visible out of the cache
+    // it looks like we can only remove old things from the cache
+    // so collect our visible tiles
+    var cachedTiles = {};
+    for (x = tileRange.minX; x <= tileRange.maxX; ++x) {
+      for (y = tileRange.minY; y <= tileRange.maxY; ++y) {
+        tileCoordKey = this.getKeyZXY(z, x, y);
+        if (this.tileCache.containsKey(tileCoordKey)) {
+          cachedTiles[tileCoordKey] = this.tileCache.get(tileCoordKey);
+        }
+      }
+    }
+
+    // clear the cache, then put our tiles back
+    this.tileCache.clear();
+    for (tileCoordKey in cachedTiles) {
+      this.tileCache.set(tileCoordKey, cachedTiles[tileCoordKey]);
+    }
+  }
+
+  // set up the tile load handler
+  var setLoadHandler = function(tileCoordKey, outOfBandTile) {
+    var onTileLoad = function() {
+      if (outOfBandTile.getState() == ol.TileState.LOADING) {
+        // listen again to get the next event
+        goog.events.listenOnce(outOfBandTile, goog.events.EventType.CHANGE,
+            onTileLoad, false, this);
+      } else if (outOfBandTile.getState() == ol.TileState.LOADED ||
+          outOfBandTile.getState() == ol.TileState.ERROR) {
+        if (this.tileCache.containsKey(tileCoordKey)) {
+          // tile loads can come in out of order. make sure we only
+          // use the most recent tile
+          var cachedTile = this.tileCache.get(tileCoordKey);
+          if (outOfBandTile.getSequenceNumber() >
+              cachedTile.getSequenceNumber()) {
+            // make sure we don't overwrite a newer out-of-band-tile
+            outOfBandTile.updateOutOfBandTile(cachedTile);
+            // replace the old tile
+            this.tileCache.replace(tileCoordKey, outOfBandTile);
+          }
+        } else {
+          // we've never seen this tile before, just add it to the cache
+          this.tileCache.set(tileCoordKey, outOfBandTile);
+        }
+        // make sure the map draws a frame after the tile is loaded
+        map.render();
```

Comment:
Tile sources should not know about the map. So I think this deserves some re-design.